### PR TITLE
Implement chat completion streaming

### PR DESCRIPTION
**Pull request by NICEXAI**

### CODE REVIEW

File: chat.go [hunk 1]
```diff
@@ -49,7 +49,7 @@ type ChatCompletionResponse struct {
 	Usage   Usage                  `json:"usage"`
 }
 
-// CreateChatCompletion — API call to Creates a completion for the chat message.
+// CreateChatCompletion — API call to Create a completion for the chat message.
```
Comment (sashabaranov) [inline chat.go:52]: thanks! 😄

File: chat_stream.go [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+	"io"
+	"net/http"
+)
+
+type ChatCompletionStreamResponse struct {
+	ID      string `json:"id"`
+	Object  string `json:"object"`
+	Created int64  `json:"created"`
+	Model   string `json:"model"`
+	Choices []struct {
```
Comment (sashabaranov) [inline chat_stream.go:17]: Can we avoid inline structs here, please?

File: chat_stream_test.go [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+import (
+	"context"
+	"errors"
+	"github.com/sashabaranov/go-gpt3/internal/test"
+	"io"
+	"testing"
+)
+
+func TestClient_CreateChatCompletionStream(t *testing.T) {
+	config := DefaultConfig(test.GetTestToken())
```
Comment (sashabaranov) [inline chat_stream_test.go:12]: Without setting the BaseURL, the test would not contact the Mock server. See the previous stream test: https://github.com/sashabaranov/go-gpt3/blob/58d99eb220f747e9712f28995d08465f77e025f3/stream_test.go#L43

Let's not bother with a mock server here and have a test for a real API. To obtain a token, please use logic similar to [here](https://github.com/sashabaranov/go-gpt3/blob/58d99eb220f747e9712f28995d08465f77e025f3/api_test.go#L14)

```go
apiToken := os.Getenv("OPENAI_TOKEN")
if apiToken == "" {
	t.Skip("Skipping testing against production OpenAI API. Set OPENAI_TOKEN environment variable to enable it.")
}
```

File: chat_test.go [hunk 4]
```diff
@@ -0,0 +1,28 @@
+package gogpt
+
+import (
+	"context"
+	"github.com/sashabaranov/go-gpt3/internal/test"
+	"testing"
+)
+
+func TestClient_CreateChatCompletion(t *testing.T) {
+	config := DefaultConfig(test.GetTestToken())
```
Comment (sashabaranov) [inline chat_test.go:10]: Same as the previous comment about the mock server

**Review by sashabaranov — CHANGES_REQUESTED** [review]

**Comment by sashabaranov** [comment]
Thank you for the awesome PR! ❤️ Let's make these minor changes happen, and I hope to merge it soon!

File: chat_stream.go (hunk 2 above)
Comment (NICEXAI) [reply chat_stream.go:17]: No problem.

Comment (NICEXAI) [reply chat_stream.go]: > Can we avoid inline structs here, please?

Completed.

File: chat_stream_test.go (hunk 3 above)
Comment (NICEXAI) [reply chat_stream_test.go]: > Without setting the BaseURL, the test would not contact the Mock server. See the previous stream test:
> 
> https://github.com/sashabaranov/go-gpt3/blob/58d99eb220f747e9712f28995d08465f77e025f3/stream_test.go#L43
> 
> Let's not bother with a mock server here and have a test for a real API. To obtain a token, please use logic similar to [here](https://github.com/sashabaranov/go-gpt3/blob/58d99eb220f747e9712f28995d08465f77e025f3/api_test.go#L14)
> 
> ```go
> apiToken := os.Getenv("OPENAI_TOKEN")
> if apiToken == "" {
> 	t.Skip("Skipping testing against production OpenAI API. Set OPENAI_TOKEN environment variable to enable it.")
> }
> ```

Sorry, I don't know much about this testing process yet. In order not to cause you any additional trouble, I temporarily removed it.

File: chat_stream.go (hunk 2 above)
Comment (sashabaranov) [reply chat_stream.go]: Thanks!

**Comment by sashabaranov** [comment]
Okay, looks good to me. Please fix linter errors, and let's merge! 🙌🏻

**Comment by NICEXAI** [comment]
> Okay, looks good to me. Please fix linter errors, and let's merge! 🙌🏻

It's done.

**Review by sashabaranov — APPROVED** [review]

**Pull request merged**